File: ros_c++/P_node.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>



int main(int argc, char* argv[])
{
    ros::init(argc,argv,"P_node");
    ros::NodeHandle n;
    ros::Publisher pub =n.advertise<std_msgs::String>("topic",10);
    ros::Rate rate(10);
    while(ros::ok())
    {
        std_msgs::String m;
        m.data="2023113135_张家源";
        pub.publish(m);
        rate.sleep();
    }
    return 0;
}

File: ros_py/P_node.py
#!/usr/bin/env python3
#coding = utf-8

import rospy
from std_msgs.msg import String
if __name__=="__main__":
    rospy.init_node("P_node")
    rospy.logwarn("node init successfully")
    pub = rospy.Publisher("topic_py",String,queue_size=10)
    rate=rospy.Rate(10)
    while(not rospy.is_shutdown()):
        m=String()
        m.data="2023113135张家源"
        pub.publish(m)  
        rate.sleep()